src/simple_ble.rs

```diff
@@ -45,7 +45,7 @@ impl BleAdvertisingDriver {
             ble_commands::ALLOW_ADVERTISMENT_BUFFER,
             advertising_buffer,
         )?;
-        shared_memory.write_bytes(&service_payload.bytes);
+        shared_memory.write_bytes(service_payload.as_ref());
```

Comment:
Good idea to use `as_ref()`! It would be even more rustic if the `as_ref` conversion was in `write_bytes`.